api/layer/notifications.go

```diff
@@ -122,10 +132,18 @@ func (n *layer) checkAndCompleteNotificationConfiguration(c *data.NotificationCo
 		return completed, errors.GetAPIError(errors.ErrNotificationTopicNotSupported)
 	}
 
+	e := prepareTestEvent(r.BucketName, r.RequestID, r.Host)
+
 	for i, q := range c.QueueConfigurations {
 		if err = checkEvents(q.Events); err != nil {
 			return
 		}
+		if err = checkRules(q.Filter.Key.FilterRules); err != nil {
+			return
+		}
+		if err = n.ncontroller.SendTestEvent(e, q.QueueArn); err != nil {
```

Comment:
I thought we discussed it on Thursday, that we can send `TestEvent` before updating configuration. I'm not sure it's bad, because, as I see, AWS checks topics before applying configuration too.

Reply:
Can we check topics without sending anything to it before updating configuration and send `TestEvent` after?